lib/improver/tests/nbhood/use_nbhood/test_CollapseMaskedNeighbourhoodCoordinate.py

```diff
@@ -0,0 +1,531 @@
+            plugin.reweight_weights(nbhooded_cube)
+
+    def test_normalizing_along_another_axis(self):
+        """Normalizing along another axis, when this is a valid thing to do.
+           This is normalizing along the rows of the input weights."""
+        input_weights = np.array([[[0.0, 0.7, 0.0, 0.0, 0.0],
+                                   [0.0, 0.3, 0.0, 0.0, 0.0],
+                                   [0.3, 0.1, 0.0, 0.0, 0.0],
+                                   [0.2, 0.0, 0.0, 0.0, 0.0],
+                                   [0.1, 0.0, 0.0, 0.0, 0.0]],
+                                  [[1.0, 0.0, 1.0, 1.0, 1.0],
+                                   [1.0, 0.0, 1.0, 1.0, 1.0],
+                                   [0.0, 0.0, 1.0, 1.0, 1.0],
+                                   [0.0, 0.0, 0.0, 0.9, 0.1],
+                                   [1.0, 1.0, 1.0, 0.0, 0.0]],
+                                  [[0.2, 0.0, 0.0, 0.0, 0.0],
+                                   [0.3, 0.0, 0.0, 0.0, 0.0],
+                                   [0.1, 0.0, 0.0, 0.0, 0.0],
+                                   [0.0, 0.0, 0.0, 0.1, 0.0],
+                                   [0.0, 0.0, 0.0, 0.4, 0.8]]])
+        expected_weights = np.array([[[0.0, 1.0, 0.0, 0.0, 0.0],
+                                      [0.0, 1.0, 0.0, 0.0, 0.0],
+                                      [0.75, 0.25, 0.0, 0.0, 0.0],
+                                      [1.0, 0.0, 0.0, 0.0, 0.0],
+                                      [1.0, 0.0, 0.0, 0.0, 0.0]],
+                                     [[0.25, 0.0, 0.25, 0.25, 0.25],
+                                      [0.25, 0.0, 0.25, 0.25, 0.25],
+                                      [0.0, 0.0, 0.333333, 0.333333, 0.333333],
+                                      [0.0, 0.0, 0.0, 0.9, 0.1],
+                                      [0.333333, 0.333333, 0.333333, 0.0, 0.0]
+                                      ],
+                                     [[1.0, 0.0, 0.0, 0.0, 0.0],
+                                      [1.0, 0.0, 0.0, 0.0, 0.0],
+                                      [1.0, 0.0, 0.0, 0.0, 0.0],
+                                      [0.0, 0.0, 0.0, 1.0, 0.0],
+                                      [0.0, 0.0, 0.0, 0.333333, 0.666667]]])
+        weights_cube = self.weights_cube.copy(input_weights)
+        nbhooded_cube = self.weights_cube.copy()
+        plugin = CollapseMaskedNeighbourhoodCoordinate(
+            "projection_x_coordinate", weights_cube)
+        plugin.reweight_weights(nbhooded_cube)
+        self.assertArrayAlmostEqual(expected_weights, weights_cube.data)
+
+
+class Test_process(IrisTest):
+
+    """Test the process method of CollapseMaskedNeighbourhoodCoordinate"""
+
+    def setUp(self):
+        """Set up a cube."""
+        self.cube = set_up_cube(
+            zero_point_indices=((0, 0, 2, 2),), num_grid_points=5)
+        self.cube = iris.util.squeeze(self.cube)
```

Comment:
As before: "self.cube" is local and docstring could be more descriptive.

Reply:
Done